### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by FrankMugagga**
### Activities
- Created a class Classroom
- Create the has-many/belongs-to relationship between Classroom and Student
- Created a class Book
- Created a class Rental 
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental
- Updated the README.md file

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @rental = date
+    @book = book
+    @person = person
```
Comment (1GORDON) [inline rental.rb]: Great job @FrankMugagga 🤝

- [ ] Kindly initialize the date instance properly within the constructor. you could do something like this `@date = date`.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  def can_use_services?
+    return true if of_age? || parent_permission
+
+    false
+  end
+
+  def add_rental(person, date)
+    Rental.new(person, self, date)
+  end
```
Comment (1GORDON) [inline person.rb]: Great job @FrankMugagga 🤝

- [ ] Kindly update your `add_rental` method. Kindly replace it with the method I have left in the screenshot below.
person-add_rentals

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @FrankMugagga ✋ 

Good job so far!

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @FrankMugagga, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**